assembly/README.md

```diff
@@ -21,15 +21,75 @@ let assembler = Assembler::default();
 let program = assembler.compile("begin push.3 push.5 add end").unwrap();
 ```
 
+## Assembler options
+By default, the assembler is instantiated in the most minimal form. To extended the capabilities of the assembler, you can apply a chain of `with_*` methods to the default instance in a builder pattern. The set of currently available options is described below.
+
+### Module provider
+To enable calls to procedures from external modules, the assembler must be supplied with a `ModuleProvider`. A module provide tells the assembler how to resolve calls to external procedures.
+
+`ModuleProvider` is a trait which can be implemented in a number of different ways. We have implemented it for the Miden [standard library](../stdlib). Thus, for example, to make Miden stdlib available to programs during compilation, the assembler can be instantiated as follows:
+
+```Rust
+use miden_assembly::Assembler;
+use miden_stdlib::StdLibrary;
+
+// instantiate the assembler with access to Miden stdlib
+let assembler = Assembler::default().with_module_provider(StdLibrary::default());
```

Comment:
Both work, but I'm actually thinking of removing `Assembler::new()` in the future since I think `Assembler::default()` captures the intent more explicitly.